src/get-activity.ts

```diff
@@ -30,31 +39,95 @@ export class GetActivity {
+      this.self,
+      this.events,
+      this.comments,
+      this.linkedReviews,
+    ]);
   }
 
   private async _getLinkedReviews(): Promise<Review[]> {
     const pulls = await collectLinkedMergedPulls(this._issueParams);
-    const promises = pulls.map((pull) => {
-      const repository = pull.source.issue.repository;
+    const promises = pulls
+      .map(async (pull) => {
+        const repository = pull.source.issue.repository;
 
-      if (!repository) {
-        throw new Error("No repository found");
-      }
-
-      const pullParams = {
-        owner: repository.owner.login,
-        repo: repository.name,
-        pull_number: pull.source.issue.number,
-      };
-      const review = new Review(pullParams);
-      return review.init().then(() => review);
-    });
+        if (!repository) {
+          console.error(`No repository found for [${pull.source.issue.repository}]`);
+          return null;
+        } else {
+          const pullParams = {
+            owner: repository.owner.login,
+            repo: repository.name,
+            pull_number: pull.source.issue.number,
+          };
+          const review = new Review(pullParams);
+          await review.init();
+          return review;
+        }
+      })
+      .filter((o) => o !== null) as Promise<Review>[];
     return Promise.all(promises);
   }
+
+  _getTypeFromComment(
+    comment: GitHubIssueComment | GitHubPullRequestReviewComment | GitHubIssue | GitHubPullRequest,
+    self: GitHubPullRequest | GitHubIssue | null
+  ) {
+    let ret = 0;
+    ret |= "pull_request_review_id" in comment ? CommentType.REVIEW : CommentType.ISSUE;
+    if (comment.user?.id === self?.user?.id) {
+      ret |= CommentType.ISSUER;
+    } else if (comment.user?.id === self?.assignee?.id) {
+      ret |= CommentType.ASSIGNEE;
+    } else if (comment.author_association === "MEMBER") {
+      ret |= CommentType.COLLABORATOR;
+    } else if (comment.author_association === "CONTRIBUTOR") {
+      ret |= CommentType.CONTRIBUTOR;
+    }
+    return ret;
+  }
+
+  get allComments() {
+    const comments: Array<
+      (GitHubIssueComment | GitHubPullRequestReviewComment | GitHubIssue | GitHubPullRequest) & { type: CommentType }
+    > = (this.comments as GitHubIssueComment[]).map((comment) => ({
+      ...comment,
+      type: this._getTypeFromComment(comment, this.self as GitHubIssue),
+    }));
+    if (this.self) {
+      comments.push({
+        ...(this.self as GitHubIssue),
+        type: this._getTypeFromComment(this.self as GitHubIssue, this.self as GitHubIssue),
+      });
+    }
+    if (this.linkedReviews) {
+      for (const linkedReview of this.linkedReviews as Review[]) {
+        if (linkedReview.self) {
+          comments.push({
+            ...(linkedReview.self as GitHubPullRequest),
+            type: this._getTypeFromComment(
+              linkedReview.self as GitHubPullRequest,
+              linkedReview.self as GitHubPullRequest
```

Comment:
I see a lot of casts in this function, are they necessary?

Reply:
`self` is of type `Promise<GitHubPullRequest> | GitHubPullRequest` so yes. But I don't know why it is typed like that, will check if we can have it only as `GitHubPullRequest`.